Dataflow analyzer throws a Class Cast Exception

**Description:**
Consider the following snippet,

```
function testFunction() {
    var testVar = table 
}
```

This throws the following class cast exception,

```
Caused by: java.lang.ClassCastException: org.wso2.ballerinalang.compiler.semantics.model.types.BType cannot be cast to org.wso2.ballerinalang.compiler.semantics.model.types.BTableType
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.getFieldNames(DataflowAnalyzer.java:765)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.checkForDuplicateKeys(DataflowAnalyzer.java:659)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.visit(DataflowAnalyzer.java:654)
	at org.wso2.ballerinalang.compiler.tree.expressions.BLangTableConstructorExpr.accept(BLangTableConstructorExpr.java:67)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.analyzeNode(DataflowAnalyzer.java:1633)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.visit(DataflowAnalyzer.java:438)
	at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:53)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.analyzeNode(DataflowAnalyzer.java:1633)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.visit(DataflowAnalyzer.java:422)
	at org.wso2.ballerinalang.compiler.tree.statements.BLangSimpleVariableDef.accept(BLangSimpleVariableDef.java:46)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.analyzeNode(DataflowAnalyzer.java:1633)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.visit(DataflowAnalyzer.java:356)
	at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:57)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.analyzeNode(DataflowAnalyzer.java:1633)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.analyzeBranch(DataflowAnalyzer.java:1615)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.visit(DataflowAnalyzer.java:347)
	at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:73)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.analyzeNode(DataflowAnalyzer.java:1633)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.visit(DataflowAnalyzer.java:289)
	at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:161)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.analyzeNode(DataflowAnalyzer.java:1633)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.DataflowAnalyzer.analyze(DataflowAnalyzer.java:270)
	at org.wso2.ballerinalang.compiler.CompilerDriver.dataflowAnalyze(CompilerDriver.java:334)
	at org.wso2.ballerinalang.compiler.CompilerDriver.compile(CompilerDriver.java:266)
	at org.wso2.ballerinalang.compiler.CompilerDriver.compilePackageSymbol(CompilerDriver.java:248)
	at org.wso2.ballerinalang.compiler.CompilerDriver.compilePackage(CompilerDriver.java:140)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:166)
	at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:137)
	at org.ballerinalang.langserver.compiler.LSModuleCompiler.compilePackagesSafe(LSModuleCompiler.java:291)
	... 11 more
```

**Affected Versions:**
SLP3 at least

--- a/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java
+++ b/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java
@@ -190,6 +190,7 @@ import org.wso2.ballerinalang.compiler.tree.types.BLangValueType;
 import org.wso2.ballerinalang.compiler.util.CompilerContext;
 import org.wso2.ballerinalang.compiler.util.Name;
 import org.wso2.ballerinalang.compiler.util.Names;
+import org.wso2.ballerinalang.compiler.util.TypeTags;
 import org.wso2.ballerinalang.compiler.util.diagnotic.BLangDiagnosticLogHelper;
 import org.wso2.ballerinalang.compiler.util.diagnotic.DiagnosticPos;
 import org.wso2.ballerinalang.util.Flags;
@@ -762,22 +763,21 @@ public class DataflowAnalyzer extends BLangNodeVisitor {
     }
 
     private List<String> getFieldNames(BLangTableConstructorExpr constructorExpr) {
-        List<String> fieldNames = ((BTableType) constructorExpr.type).fieldNameList;
-
-        if (fieldNames != null) {
-            return fieldNames;
+        List<String> fieldNames = null;
+        if (constructorExpr.type.tag == TypeTags.TABLE) {
+            fieldNames = ((BTableType) constructorExpr.type).fieldNameList;
+            if (fieldNames != null) {
+                return fieldNames;
+            }
         }
-
         if (constructorExpr.tableKeySpecifier != null &&
                 !constructorExpr.tableKeySpecifier.fieldNameIdentifierList.isEmpty()) {
             BLangTableKeySpecifier tableKeySpecifier = constructorExpr.tableKeySpecifier;
-            fieldNames = tableKeySpecifier.fieldNameIdentifierList.stream().map(identifier ->
+            return tableKeySpecifier.fieldNameIdentifierList.stream().map(identifier ->
                     ((BLangIdentifier) identifier).value).collect(Collectors.toList());
         } else {
             return new ArrayList<>();
         }
-
-        return fieldNames;
     }
 
     @Override

